src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs

```diff
@@ -34,19 +35,22 @@ internal class ExceptionHandlerMiddlewareImpl
     /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> used for logging.</param>
     /// <param name="options">The options for configuring the middleware.</param>
     /// <param name="diagnosticListener">The <see cref="DiagnosticListener"/> used for writing diagnostic messages.</param>
+    /// <param name="exceptionHandlers"></param>
     /// <param name="problemDetailsService">The <see cref="IProblemDetailsService"/> used for writing <see cref="ProblemDetails"/> messages.</param>
     public ExceptionHandlerMiddlewareImpl(
         RequestDelegate next,
         ILoggerFactory loggerFactory,
         IOptions<ExceptionHandlerOptions> options,
         DiagnosticListener diagnosticListener,
+        IEnumerable<IExceptionHandler> exceptionHandlers,
         IProblemDetailsService? problemDetailsService = null)
     {
         _next = next;
         _options = options.Value;
         _logger = loggerFactory.CreateLogger<ExceptionHandlerMiddleware>();
         _clearCacheHeadersDelegate = ClearCacheHeaders;
         _diagnosticListener = diagnosticListener;
+        _exceptionHandlers = exceptionHandlers;
```

Comment:
Turn this into a list or array so we can avoid the extra enumerator allocation on exceptions.

Reply:
See https://github.com/dotnet/runtime/blob/953d290e756d29707b9829e3ea00ef8f2fc31a9f/src/libraries/Microsoft.Extensions.Options/src/OptionsFactory.cs#L38-L45

Reply:
Done!